plutus-tx-plugin/test/IsData/deconstructData.pir.golden

```diff
@@ -190,7 +190,19 @@
                             )
                           )
                         ]
-                        (abs dead (type) { Nothing [ [ Tuple2 a ] b ] })
+                        (abs
+                          dead
+                          (type)
+                          (let
+                            (nonrec)
+                            (termbind
+                              (strict)
+                              (vardecl ds [ Maybe b ])
```

Comment:
Something funny is going on here. This is dead - we're not removing it because the binding is now strict. But it's weird that it's dead in the input program!

Reply:
I think this is from `Nothing <*> _ = Nothing`, and it is the same situation as `||` and `&&`. The second argument should always be strictly evaluated, but when we didn't have `Strict`, GHC may unconditionally inline `<*>` which avoided evaluating the second argument.

This can be quite bad, and may be a main contributor of some of the increased budgets :neutral_face: